### PR TITLE
Implement new version chooser frontend, tweak backend

### PR DESCRIPTION
**Pull request by Williangalvani**
fix chooser part of #779 

everything works, I'm now working on simplifying the interface for regular users.

opening as a draf as I"m having issues running our hooks locally.

image

### Update to latest of a custom tag:
image

### Update from a Beta version to another beta or to a stable (we have none so far):
image

to-do:
 - [x] #522
 - [x] #338 
 - [x] #519
 - [x] #267 (sha is hidden unless in Pirate Mode) 

image is up at williangalvani/companion-core:version-chooser-updates

### CODE REVIEW

File: core/frontend/src/components/version-chooser/VersionChooser.vue [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+          this.waiting = true
+        })
+    },
+    fixVersion(version: string) {
+      /** It turned out that our semvers are wrong... oopss
+      This turns 1.0.0.beta12 into 1.0.0-beta.12
+      Additionally filters out tags with no '.' in it, which
+      can be improved
+      */
+      if (version.includes('beta') && !version.includes('-beta.')) {
```
Comment (patrickelectric) [inline core/frontend/src/components/version-chooser/VersionChooser.vue:243]:
```suggestion
      if (version.match(/.beta\d+/)) {
```

File: core/frontend/src/components/version-chooser/VersionChooser.vue [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+      if (!version.includes('.')) {
+        return null
+      }
+      return version
+    },
+    isSemVer(version: string): boolean {
+      // validates a version as SemVer compliant
+      try {
+        const fixed_version = this.fixVersion(version)
+        if (fixed_version == null) {
```
Comment (patrickelectric) [inline core/frontend/src/components/version-chooser/VersionChooser.vue:255]: This check is unnecessary from what I can understand reading the code, the `new SemVer` should raise an error and return false

File: core/frontend/src/components/version-chooser/VersionChooser.vue [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+    >
+      <spinning-logo
+        size="100%"
+      />
+    </v-overlay>
+  </div>
+</template>
+
+<script lang="ts">
+import { gt, SemVer } from 'semver'
```
Comment (patrickelectric) [inline core/frontend/src/components/version-chooser/VersionChooser.vue:174]:
```suggestion
import { gt as sem_ver_greater, SemVer } from 'semver'
```

File: core/frontend/src/components/version-chooser/VersionChooser.vue [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+          }
+          return gt(new SemVer(ver_a), new SemVer(ver_b)) === true ? -1 : 1
+        },
+      )
+    },
+    runningBeta() {
+      if (this.current_version === null) {
+        return false
+      }
+      return this.current_version.tag.includes('beta')
```
Comment (patrickelectric) [inline core/frontend/src/components/version-chooser/VersionChooser.vue:283]: `return this.current_version.?tag.?includes('beta') ?? false`

File: core/frontend/src/components/version-chooser/VersionChooser.vue [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+      const ordered_list = this.sortVersions(
+        this.available_versions.remote.map((image) => image.tag)
+          .filter((tag) => this.isSemVer(tag) && tag.includes('beta')),
+      )
+      return ordered_list ? ordered_list[0] : undefined
+    },
+    getLatestStable() {
+      const ordered_list = this.sortVersions(
+        this.available_versions.remote.map((image) => image.tag)
+          .filter((tag) => this.isSemVer(tag) && !tag.includes('beta')),
```
Comment (patrickelectric) [inline core/frontend/src/components/version-chooser/VersionChooser.vue:313]: It says 'beta' here, we should make a function to encapsulate the map logic and receive an string as parameters, to use for beta and stable

File: core/frontend/src/components/version-chooser/VersionChooser.vue (hunk 2 above)
Comment (Williangalvani) [reply core/frontend/src/components/version-chooser/VersionChooser.vue:255]: typescript won't allow 'null' as a paraeter for SemVer(), that's why the check is there

File: core/frontend/src/components/version-chooser/VersionChooser.vue (hunk 1 above)
Comment (Williangalvani) [reply core/frontend/src/components/version-chooser/VersionChooser.vue:243]: in hindsight, `version.includes('.beta')` should be enough here.

**Review by patrickelectric — APPROVED** [review]
image

**Pull request merged**